File: 20250330/leetcode2109. 向字符串添加空格/main.cpp
#include <iostream>
#include <string>
#include <unordered_set>
#include <vector>

using namespace std;

class Solution {
public:
  string addSpaces(string s, vector<int> &spaces) {
    int n = s.size();
    unordered_set<int> set = {spaces.begin(), spaces.end()};
    string ans;
    for (int i = 0; i < n; i++) {
      if (set.count(i)) {
        ans.push_back(' ');
      }
      ans.push_back(s[i]);
    }
    return ans;
  }
};

File: 20250330/leetcode2109. 向字符串添加空格/main.js
/*
 * @Author: mangwu                                                             *
 * @File: main.js                                                              *
 * @Date: 2025-03-30 21:26:32                                                  *
 * @LastModifiedDate: 2025-03-30 21:37:48                                      *
 * @ModifiedBy: mangwu                                                         *
 * -----------------------                                                     *
 * Copyright (c) 2025 mangwu                                                   *
 * -----------------------                                                     *
 * @HISTORY:                                                                   *
 * Date   	            By 	    Comments                                       *
 * ---------------------	--------	----------------------------------------------- *
 */

/**
 * @param {string} s
 * @param {number[]} spaces
 * @return {string}
 */
var addSpaces = function (s, spaces) {
  const n = s.length;
  const ans = [];
  const set = new Set(spaces);
  for (let i = 0; i < n; i++) {
    if (set.has(i)) ans.push(" ");
    ans.push(s[i]);
  }
  return ans.join("");
};
